### PR TITLE
fix(forks): wrap `defines` to support `undefined` values

### PR DESCRIPTION
**Pull request by AriPerkkio**
### Description

- Fixes https://github.com/vitest-dev/vitest/issues/5273

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [x] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [x] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/pools/forks.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import type { PoolProcessOptions, ProcessPool, RunWithFiles } from '../pool'
 import type { WorkspaceProject } from '../workspace'
 import { envsOrder, groupFilesByEnv } from '../../utils/test-helpers'
 import { groupBy, resolve } from '../../utils'
+import { wrapForksConfig } from '../../runtime/workers/utils'
```
Comment (AriPerkkio) [inline packages/vitest/src/node/pools/forks.ts:12]: It might be bad to import from `runtime/workers` on the main thread's side. Any better place for this?

Comment (AriPerkkio) [reply packages/vitest/src/node/pools/forks.ts:12]: `utils/config-helpers.ts` looks better.

File: packages/vitest/src/runtime/workers/utils.ts [hunk 2]
```diff
@@ -27,16 +25,3 @@ export function createForksRpcOptions(nodeV8: typeof import('v8')): WorkerRpcOpt
     },
   }
 }
-
-function parsePossibleRegexp(str: string | RegExp) {
```
Comment (sheremet-va) [inline packages/vitest/src/runtime/workers/utils.ts:31]: I would be cautious to mix thread level utils and workers utils, rollup can bundle them together one time and we won't notice

Comment (AriPerkkio) [reply packages/vitest/src/runtime/workers/utils.ts:31]: Yep I agree. I'll refactor the d7c319e4af94cac7a1361523c989aa6f0b13476f once more.

**Pull request merged**